students/DiannaTingg/lessons/lesson03/assignment/test_basic_operations.py

```diff
@@ -0,0 +1,190 @@
+"""
+Tests for basic_operations
+"""
+
+import peewee as pw
+import pytest
+import customer_model as cm
+import basic_operations as bo
+
+customer1 = {"customer_id": "1", "first_name": "Fred", "last_name": "Flintstone",
+             "home_address": "301 Cobblestone Way, Bedrock, 70777", "phone_number": "928-635-2600",
+             "email_address": "fred@hotmail.com", "active_status": True, "credit_limit": 100.00}
+
+customer2 = {"customer_id": "2", "first_name": "Barney", "last_name": "Rubble",
+             "home_address": "303 Cobblestone Way, Bedrock, 70777", "phone_number": "928-635-2700",
+             "email_address": "barney@aol.com", "active_status": False, "credit_limit": 50.00}
+
+customer3 = {"customer_id": "3", "first_name": "Wilma", "last_name": "Flintstone",
+             "home_address": "301 Cobblestone Way, Bedrock, 70777", "phone_number": "928-635-2600",
+             "email_address": "wilma@hotmail.com", "active_status": True, "credit_limit": 100.00}
+
+customer4 = {"customer_id": "4", "first_name": "Betty", "last_name": "Rubble",
+             "home_address": "303 Cobblestone Way, Bedrock, 70777", "phone_number": "928-635-2700",
+             "email_address": "barney@aol.com", "active_status": False, "credit_limit": 50.00}
+
+
+def clear_database():
+    cm.database.drop_tables([
+        cm.Customer
+    ])
+
+    cm.database.close()
+
+
+def create_empty_database():
+    clear_database()
+
+    cm.database.create_tables([
+        cm.Customer
+    ])
+
+    cm.database.close()
+
+
+def test_add_customers():
+    create_empty_database()
+    bo.add_customer(**customer1)
+    bo.add_customer(**customer2)
+
+    test_customer1 = cm.Customer.get(cm.Customer.customer_id == "1")
+    assert test_customer1.first_name == customer1["first_name"]
+    assert test_customer1.last_name == customer1["last_name"]
+
+    test_customer2 = cm.Customer.get(cm.Customer.customer_id == "2")
+    assert test_customer2.first_name == customer2["first_name"]
+    assert test_customer2.last_name == customer2["last_name"]
+
+    clear_database()
+
+
+def test_add_customers_duplicate():
+    create_empty_database()
+    bo.add_customer(**customer1)
+
+    with pytest.raises(pw.IntegrityError):
+        bo.add_customer(**customer1)
+
+    clear_database()
```

Comment:
Great. Easy to forget the tests looking for errors. You might consider raising a more friendly for your user error instead (although I ran into problems with testing doing this with peewee).